server/fleet/mdm_test.go

```diff
@@ -333,6 +368,67 @@ func TestMDMProfileSpecUnmarshalJSON(t *testing.T) {
 		require.Equal(t, "oldpath", p.Path)
 		require.Empty(t, p.Labels)
 	})
+
+	t.Run("changing labels", func(t *testing.T) {
```

Comment:
This was a released bug found during testing. I opted to fix this one, but there are a few others that I'm planning to file as separate issues to keep this feature branch moving forward.